client/state/themes/selectors.js

```diff
@@ -0,0 +1,200 @@
+/** @ssr-ready **/
+
+/**
+ * External dependencies
+ */
+import { some, split } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import { getSiteSlug, getSiteOption, isJetpackSite } from 'state/sites/selectors';
+import { getSitePurchases } from 'state/purchases/selectors';
+import { isPremiumTheme, oldShowcaseUrl } from './utils';
+
+/**
+ * Returns the URL for a given theme's details sheet.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Object}  theme  Theme object
+ * @param  {?Number} siteId Site ID to optionally use as context
+ * @return {?String}        Theme details sheet URL
+ */
+export function getThemeDetailsUrl( state, theme, siteId ) {
+	if ( ! theme ) {
+		return null;
+	}
+
+	if ( isJetpackSite( state, siteId ) ) {
+		return getSiteOption( state, siteId, 'admin_url' ) + 'themes.php?theme=' + theme.id;
+	}
+
+	let baseUrl = oldShowcaseUrl + theme.id;
+	if ( config.isEnabled( 'manage/themes/details' ) ) {
+		baseUrl = `/theme/${ theme.id }`;
+	}
+
+	return baseUrl + ( siteId ? `/${ getSiteSlug( state, siteId ) }` : '' );
+}
+
+/**
+ * Returns the URL for a given theme's setup instructions
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Object}  theme  Theme object
+ * @param  {?Number} siteId Site ID to optionally use as context
+ * @return {?String}        Theme setup instructions URL
+ */
+export function getThemeSupportUrl( state, theme, siteId ) {
+	if ( ! theme || ! isPremiumTheme( theme ) ) {
+		return null;
+	}
+
+	const sitePart = siteId ? `/${ getSiteSlug( state, siteId ) }` : '';
+
+	if ( config.isEnabled( 'manage/themes/details' ) ) {
+		return `/theme/${ theme.id }/setup${ sitePart }`;
+	}
+
+	return `${ oldShowcaseUrl }${ sitePart }${ theme.id }/support`;
+}
+
+/**
+ * Returns the URL for a given theme's support page.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Object}  theme  Theme object
+ * @param  {?Number} siteId Site ID to optionally use as context
+ * @return {?String}        Theme support page URL
+ */
+export function getThemeHelpUrl( state, theme, siteId ) {
+	if ( ! theme ) {
+		return null;
+	}
+
+	if ( isJetpackSite( state, siteId ) ) {
+		return '//wordpress.org/support/theme/' + theme.id;
```

Comment:
Will there always be a theme at this URL? Or is it too early to think about details such details? :D